src/main/java/campus/infrastructure/Parser.java

```diff
@@ -0,0 +1,217 @@
+package campus.infrastructure;
+
+import campus.exceptions.CampusException;
+import campus.tasks.Deadline;
+import campus.tasks.Event;
+import campus.tasks.Task;
+import campus.tasks.ToDos;
+
+
+/**
+ * Parser handles all the logic for when a command is entered into the ChatBot, it serves as the bridge between the
+ * storage object and taskList object and is the main handler for data travelling between these two fields
+ */
+public class Parser {
+    private final Ui ui;
+    private TaskList taskList;
+    private final Storage storage;
+
+    /**
+     * Initialiser for Paser class
+     * @param ui The Ui Class
+     * @param taskList The TaskList Class
+     * @param storage The Storage Class
+     */
+    public Parser(Ui ui, TaskList taskList, Storage storage) {
+        this.ui = ui;
+        this.storage = storage;
+
+        try {
+            this.taskList = taskList;
+            this.taskList.updateListFromFile(storage.getListOfStrings());
+        } catch (CampusException e) {
+            this.ui.displayErrorMessage(e);
+        }
+    }
+
+    /**
+     * Responds on user input
+     */
+    public String respond(String userInput) {
+        try {
+            return this.processCommand(userInput);
+        } catch (CampusException e) {
+            return this.ui.displayErrorMessage(e);
+        }
+    }
```

Comment:
Consider providing some clearer naming for this method. Maybe something like "executeUserInput"?